### PR TITLE
ci: setup model service ci

### PR DESCRIPTION
**Pull request by dgauldie**

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    name: build-bake-push
+    runs-on: ubuntu-22.04
+    if: github.ref_name == 'main' || contains(github.event.ref, '/tags/')
+    permissions:
+      packages: write
+      contents: read
+
+    # this job depends on the following jobs before beginning
+    needs:
+      - targets
```
Comment (Nithos) [inline .github/workflows/publish.yaml:55]: Perhaps add a TODO here. If there are any test workflows they should be invoked above similarly to `targets` and dependent on here. This way tests are always run before the build. Those workflows should be made reusable and would just require a `uses` clause (see Terrarium for examples)

**Review by Nithos — APPROVED** [review]
LGTM
Small comment on future addition

File: .github/workflows/commitlint.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      # Uncomment this if you want to validate all commits in this PR not just the PR comment
+      # with:
+      #   fetch-depth: 0
+      - name: Use Node.js 18.8.0
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18.8.0
+          cache: 'yarn'
```
Comment (mwdchang) [inline .github/workflows/commitlint.yaml:24]: What does the node stuff do?

Comment (Nithos) [reply .github/workflows/commitlint.yaml:24]: needed for yarn below to install the conventional commit

Comment (Nithos) [reply .github/workflows/commitlint.yaml:24]: @dgauldie remove the `cache` part... there is no lock file here so can't be cached

File: .github/workflows/publish.yaml (hunk 1 above)
Comment (dgauldie) [reply .github/workflows/publish.yaml]: fixed

**Pull request merged**